File: CS/CS_Group2/Book.cpp
#include <iostream>
#include <fstream>
#include <vector> //for precised arrays.
#include <iomanip>
#include <string>
#include <cstdlib>
#include <limits>
#include <algorithm> // to find the value in range
using namespace std;

class Book {
private:
    string title;
    string author;
    int id;
    bool isAvailable;
public:
    Book(string t, string a, int i) {
        title = t;
        author = a;
        id = i;
        isAvailable = true;
    }

    // Getters and Setters
    string getTitle() { return title; }
    string getAuthor() { return author; }
    int getId() { return id; }
    bool getAvailability() { return isAvailable; }
    void setAvailability(bool availability) { isAvailable = availability; }

    void displayBook() {
        cout << setw(5) << id << setw(20) << title << setw(20) << author
             << setw(20) << (isAvailable ? "Available" : "Borrowed") << endl;
    }
};

File: Acity_JAVA_Project/Warren Ojbonnaya  ESONU/Book.java

public class Book {
    private String title;
    private String author;
    private int id;
    private boolean isAvailable;

    public Book(String title, String author, int id) {
        this(title, author, id, true);
    }

    public Book(String title, String author, int id, boolean isAvailable) {
        this.title = title;
        this.author = author;
        this.id = id;
        this.isAvailable = isAvailable;
    }

    // Getters and setters
    public String getTitle() {
        return title;
    }

    public String getAuthor() {
        return author;
    }

    public int getId() {
        return id;
    }

    public boolean getAvailability() {
        return isAvailable;
    }

    public void setAvailable(boolean available) {
        isAvailable = available;
    }

    @Override
    public String toString() {
        return String.format("%d: %s by %s (%s)",
                id, title, author, isAvailable ? "Available" : "Borrowed");
    }
}